src/catch2/internal/catch_textflow.cpp

```diff
@@ -26,117 +26,225 @@ namespace {
         return std::memchr( chars, c, sizeof( chars ) - 1 ) != nullptr;
     }
 
-    bool isBoundary( std::string const& line, size_t at ) {
-        assert( at > 0 );
-        assert( at <= line.size() );
-
-        return at == line.size() ||
-               ( isWhitespace( line[at] ) && !isWhitespace( line[at - 1] ) ) ||
-               isBreakableBefore( line[at] ) ||
-               isBreakableAfter( line[at - 1] );
-    }
-
 } // namespace
 
 namespace Catch {
     namespace TextFlow {
+        void AnsiSkippingString::preprocessString() {
+            for ( auto it = m_string.begin(); it != m_string.end(); ) {
+                // try to read through an ansi sequence
+                while ( it != m_string.end() && *it == '\033' &&
+                        it + 1 != m_string.end() && *( it + 1 ) == '[' ) {
```

Comment:
Probably better as

```suggestion
                while ( it + 1 < m_string.end() && *it == '\033' &&
                         *( it + 1 ) == '[' ) {
```
(with the two bounds check merged into 1)

Reply:
I tried merging the two bound checks into one however MSVC's STL was not very happy and aborted due to attempting to seek the iterator past the end.

Reply:
Oh. It could still be done with single check (comparing the distance to 2), but the readability improvement would be lost.

nevermind then